src/AutoRest.CSharp/MgmtTest/Generation/ScenarioTestWriter.cs

```diff
@@ -0,0 +1,347 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using AutoRest.CSharp.Generation.Writers;
+using AutoRest.CSharp.Mgmt.AutoRest;
+using AutoRest.CSharp.Input;
+using AutoRest.CSharp.Mgmt.Output;
+using AutoRest.CSharp.Mgmt.Models;
+using AutoRest.CSharp.Utilities;
+using AutoRest.CSharp.Common.Generation.Writers;
+using Azure.ResourceManager.Resources;
+
+namespace AutoRest.CSharp.MgmtTest.Generation
+{
+    /// <summary>
+    /// Code writer for resource test.
+    /// </summary>
+    internal class ScenarioTestWriter: ClientWriter
+    {
+        protected TestDefinitionModel testDef;
+        protected CodeWriter writer;
+        protected string TestNamespace => MgmtContext.Context.DefaultNamespace + ".Tests.Scenario";
+        public string TypeName => System.IO.Path.GetFileNameWithoutExtension(testDef.FilePath).ToCleanName();
+        protected string TestBaseName => $"ManagementRecordedTestBase<{MgmtContext.Context.DefaultLibraryName}TestEnvironment>";
+
+        public ScenarioTestWriter(CodeWriter writer, TestDefinitionModel testDef)
+        {
+            this.testDef = testDef;
+            this.writer = writer;
+        }
+
+        public void WriteScenarioTest()
+        {
+            WriteUsings(writer);
+
+            using (writer.Namespace(TestNamespace))
+            {
+                writer.WriteXmlDocumentationSummary($"Test generated from Api Scenario {testDef.FilePath}");
+                writer.Append($"public partial class {TypeName:D} : ");
+                writer.Line($"{TestBaseName}");
+                using (writer.Scope())
+                {
+                    WriteTesterCtors();
+
+                    WritePrepareMethod();
+                    WriteCleanUpMethod();
+
+                    foreach (var scenario in testDef.Scenarios)
+                    {
+                        WriteScenarioMethod(scenario);
+                    }
+                }
+            }
+        }
+
+        protected void WriteUsings(CodeWriter writer)
+        {
+            writer.UseNamespace("System");
+            writer.UseNamespace("System.Threading.Tasks");
+            writer.UseNamespace("System.Net");
+            writer.UseNamespace("System.Collections.Generic");
+            writer.UseNamespace("Azure.Core.TestFramework");
+            writer.UseNamespace("Azure.ResourceManager.TestFramework");
```

Comment:
Some of the namespaces here are redundant
```suggestion
            writer.UseNamespace("Azure.Core.TestFramework");
            writer.UseNamespace("Azure.ResourceManager.TestFramework");
```